src/Repositories/OrgDataRepository.cs

```diff
@@ -0,0 +1,136 @@
+﻿using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Internal.Query;
+using JsonApiDotNetCore.Serialization;
+using JsonApiDotNetCore.Services;
+using Microsoft.Extensions.Logging;
+using SIL.Transcriber.Models;
+using SIL.Transcriber.Services;
+using SIL.Transcriber.Utility;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using static SIL.Transcriber.Utility.Extensions.JSONAPI.FilterQueryExtensions;
+
+namespace SIL.Transcriber.Repositories
+{
+    public class OrgDataRepository : BaseRepository<OrgData>
+    {
+        protected readonly IJsonApiSerializer jsonApiSerializer;
+        protected readonly OrganizationService organizationService;
+        protected readonly GroupMembershipService gmService;
+        protected readonly CurrentUserRepository currentUserRepository;
+
+        public OrgDataRepository(
+              ILoggerFactory loggerFactory,
+              IJsonApiContext jsonApiContext,
+              CurrentUserRepository CurrentUserRepository,
+              IDbContextResolver contextResolver,
+              IJsonApiSerializer jsonSer,
+              OrganizationService orgService,
+              GroupMembershipService grpMemService
+          ) : base(loggerFactory, jsonApiContext, CurrentUserRepository, contextResolver)
+        {
+            jsonApiSerializer = jsonSer;
+            currentUserRepository = CurrentUserRepository;
+            organizationService = orgService;
+            gmService = grpMemService;
+        }
+
+        private bool CheckAdd(int check, object entity, DateTime dtBail,  int start, ref int completed, ref string data)
+        {
+            Debug.WriteLine(check.ToString() + ":" + DateTime.Now.ToString() + dtBail.ToString());
+            if (DateTime.Now > dtBail) return false;
+            if (start <= check)
+            {
+                string thisdata = jsonApiSerializer.Serialize(entity);
+                if (data.Length + thisdata.Length > (1000000 * 5.5))
+                    return false;
+                data += (check > start ?  "," : "") + thisdata;
```

Comment:
so on line 45, we check for start <= check. Here we test for check > start?

Reply:
perhaps it would be clearer if I changed it to check for the == case...